mitodl/mit-open
# Applies new fixes for the homepage and unit page testimonial sliders

### What are the relevant tickets?

Closes mitodl/hq#4644
Closes mitodl/hq#4622

### Description (What does it do?)

Implements a bunch of styling changes to bring the testimonials inline with the Figma designs. There's a lot of them but the gist of it is:
- We now break at `md` and `sm` so there's different styling for tablet users and mobile/phone users
- Testimonial text is truncated and clamped to a specific number of lines/characters per breakpoint
- On the homepage, we eliminate the marketing card image at `md` or lower
- On the homepage, we eliminate the attestant avatar at `sm` or lower
- The avatar image is now _optional_ - we use a Remixicon icon if there's no avatar
- The homepage attestation text has a unique font size - it doesn't match anything we're using currently

Here's some tables to summarize some of the changes as well. On the unit page:
| | `sm` | `md` | `lg` |
|---|---|---|---|
| Attestation max chars | 350 | 350 | 350 |
| Attestation clamp lines | 8 | 6 | 4 |
| Attestation font | `subtitle2` | `subtitle1` | `h5` |
| Attestant title clamp lines | 2 | 2 | 2 |

On the homepage:
| | `sm` | `md` | `lg` |
|---|---|---|---|
| Attestation max chars | 350 | 350 | 350 |
| Attestation clamp lines | 11 | 7 | 7 |
| Attestation font | `subtitle1` | 20px | 20px |
| Attestant title clamp lines | 2 | 2 | 2 |
| Attestant avatar | _No_ | Yes | Yes |

### Screenshots (if appropriate):

*Homepage*

Desktop: long attestation, long attestant title
![Screenshot 2024-06-20 at 2 34 46 PM](https://github.com/mitodl/mit-open/assets/945611/6ba9cb43-e261-4e78-b5c2-276404872ee7)

Desktop: no avatar
![Screenshot 2024-06-20 at 2 34 27 PM](https://github.com/mitodl/mit-open/assets/945611/5167980a-2673-48df-8941-b480ac66d204)

Tablet: long attestation, long attestant title
![Screenshot 2024-06-20 at 2 35 29 PM](https://github.com/mitodl/mit-open/assets/945611/c6e3e78e-534b-4cbc-923b-214e026b7a6e)

Tablet: long attestation, no avatar
![Screenshot 2024-06-20 at 2 37 25 PM](https://github.com/mitodl/mit-open/assets/945611/3ff09d37-19af-41d7-913b-0bfd5133789d)

Mobile:
![Screenshot 2024-06-20 at 2 39 30 PM](https://github.com/mitodl/mit-open/assets/945611/8e0fa043-4f2f-45b3-9f7e-02747dabe845)

*Unit Page*

Desktop: long attestation, long attestant title
![Screenshot 2024-06-20 at 2 42 54 PM](https://github.com/mitodl/mit-open/assets/945611/dee88e9f-bc39-48d2-b062-2d868f62af5d)

Desktop: no avatar
![Screenshot 2024-06-20 at 2 42 48 PM](https://github.com/mitodl/mit-open/assets/945611/59d77e48-ff9a-4c66-91b6-dddc3aa990de)

Tablet: long attestation, long attestant title
![Screenshot 2024-06-20 at 2 42 00 PM](https://github.com/mitodl/mit-open/assets/945611/c0a22a85-9a22-4e90-824f-a650951684a1)

Tablet: no avatar:
![Screenshot 2024-06-20 at 2 42 18 PM](https://github.com/mitodl/mit-open/assets/945611/04e78c01-86f3-4d7b-8e48-abd67b3d27c9)

Mobile: long attestation, long attestant title
![Screenshot 2024-06-20 at 2 41 25 PM](https://github.com/mitodl/mit-open/assets/945611/cf904ff2-042a-42cd-953f-9ed7a0d35aeb)

Mobile: no avatar
![Screenshot 2024-06-20 at 2 41 05 PM](https://github.com/mitodl/mit-open/assets/945611/e8d9ed7b-f4d3-4b30-8bc6-c33ba510e3cd)

### How can this be tested?

Automated tests should pass. 

You should be able to configure an attestation with no avatar image set. In addition, you should be able to clear avatars from attestations. 

The design should look like the design in Figma. (Links are available in the original issues.)
